test/devicecontroller/DefaultDeviceController.test.ts

```diff
@@ -2329,6 +2350,54 @@ describe('DefaultDeviceController', () => {
       await delay(100);
       expect(called).to.be.true;
     });
+
+    it('clears the cache if it has empty label when setting new device trigger', async () => {
+      deviceController.setDeviceLabelTrigger(async () => new MediaStream());
+
+      // This will establish the cache
+      const devices: MediaDeviceInfo[] = await deviceController.listAudioInputDevices();
+
+      // @ts-ignore
+      expect(deviceController.deviceInfoCache).to.not.be.null;
+
+      let hasDeviceLabels = true;
+      for (const device of devices) {
+        if (!device.label) {
+          hasDeviceLabels = false;
+          break;
+        }
+      }
```

Comment:
Use a simpler one-line phrasing and/or break out and use a helper function, please.

```typescript
function anyDevicesAreUnlabeled(devices: MediaDeviceInfo[]): boolean {
  return devices.some(device => !device.label);
}
```

Reply:
Addressed.